### PR TITLE
Report helpful error if filter query string malformed

### LINKED ISSUE #1230: filter: Improperly handled `ValueError: not enough values to unpack (expected 2, got 1)` in ` _parse_filter_query`
### Current Behavior
I get an error that's not very helpful:

```
        augur filter             --sequences data/sequences.fasta.zst             --metadata results/metadata.tsv             --exclude config/exclude_accessions_mpxv.txt             --output-sequences results/hmpxv1/good_sequences.fasta.zst             --output-metadata results/hmpxv1/good_metadata.tsv             --min-date 2017             --min-length 180000             --exclude-where outbreak!=hMPXV-1 "coverage<0.95" "divergence<550" "missing_data>10000" "divergence>650" QC_rare_mutations!=good             --output-log results/hmpxv1/good_filter.log
        
Note: You did not provide a sequence index, so Augur will generate one. You can generate your own index ahead of time with `augur index` and pass it with `augur filter --sequence-index`.
Traceback (most recent call last):
  File "/opt/homebrew/Caskroom/miniforge/base/envs/py11/lib/python3.11/site-packages/augur/__init__.py", line 66, in run
    return args.__command__.run(args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniforge/base/envs/py11/lib/python3.11/site-packages/augur/filter/__init__.py", line 97, in run
    return _run(args)
           ^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniforge/base/envs/py11/lib/python3.11/site-packages/augur/filter/_run.py", line 192, in run
    seq_keep, sequences_to_filter, sequences_to_include = apply_filters(
                                                          ^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniforge/base/envs/py11/lib/python3.11/site-packages/augur/filter/include_exclude_rules.py", line 707, in apply_filters
    passed = metadata.pipe(
             ^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniforge/base/envs/py11/lib/python3.11/site-packages/pandas/core/generic.py", line 5839, in pipe
    return com.pipe(self, func, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniforge/base/envs/py11/lib/python3.11/site-packages/pandas/core/common.py", line 513, in pipe
    return func(obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniforge/base/envs/py11/lib/python3.11/site-packages/augur/filter/include_exclude_rules.py", line 139, in filter_by_exclude_where
    column, op, value = _parse_filter_query(exclude_where)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Caskroom/miniforge/base/envs/py11/lib/python3.11/site-packages/augur/filter/include_exclude_rules.py", line 100, in _parse_filter_query
    column, value = re.split(r'!?=', query)
    ^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected 2, got 1)


An error occurred (see above) that has not been properly handled by Augur.
To report this, please open a new issue including the original command and the error above:
    <https://github.com/nextstrain/augur/issues/new/choose>
```
### Expected behavior

Error is properly handled and error message tells me how to fix

### System

Augur 22.0.1, Python 3.11 on ARM (!) macOS

### PR DESCRIPTION
**Pull request by corneliusroemer**
Resolves #1230
filter: Improperly handled `ValueError: not enough values to unpack (expected 2, got 1)` in ` _parse_filter_query`

### Description of proposed changes
Validate, and raise error with helpful warning if the query string is malformed, rather than raising unhelpful Value Error

New error output:

```bash
❌1 ❯ augur filter --exclude-where "coverage<0.95"   --sequences data/sequences.fasta.zst   --metadata results/metadata.tsv   --exclude config/exclude_accessions_mpxv.txt  --output-sequences results/hmpxv1/good_sequences.fasta.zst  --output-metadata results/hmpxv1/good_metadata.tsv   --min-length 180000   
Note: You did not provide a sequence index, so Augur will generate one. You can generate your own index ahead of time with `augur index` and pass it with `augur filter --sequence-index`.
ERROR: Invalid query `coverage<0.95`. Hint: Queries must follow the pattern of `property=value` or `property!=value`.
```

### Testing
What steps should be taken to test the changes you've proposed?
If you added or changed behavior in the codebase, did you update the tests, or do you need help with this?

- [ ] Checks pass

### Checklist

- [ ] Add a message in [CHANGES.md](https://github.com/nextstrain/augur/blob/HEAD/CHANGES.md) summarizing the changes in this PR that are end user focused. Keep headers and formatting consistent with the rest of the file.

### CODE REVIEW

File: augur/filter/include_exclude_rules.py [hunk 1]
```diff
@@ -97,10 +97,11 @@ def _parse_filter_query(query):
     ('property', <built-in function ne>, 'value')
 
     """
-    column, value = re.split(r'!?=', query)
-    op = operator.eq
-    if "!=" in query:
-        op = operator.ne
+    split_query = re.split(r'!?=', query)
+    if len(split_query) != 2:
+        raise AugurError(f"Invalid query `{query}`. Hint: Queries must follow the pattern of `property=value` or `property!=value`.")
```
Comment (corneliusroemer) [inline augur/filter/include_exclude_rules.py:102]:
```suggestion
        raise AugurError(f"Invalid filter query `{query}`. Hint: Queries must follow the pattern of `property=value` or `property!=value`.")
```

**Comment by huddlej** [comment]
@corneliusroemer The exclude-where/include-where syntax was never designed to be a full-featured query syntax, but you could use the `--query` interface which supports pandas-style query syntax to select the records you want. Instead of "excluding where" `coverage<0.95`, you would rephrase the query as a filter like `--query "coverage >= 0.95"`.

I agree though that the error message could be improved! As part of the hint here, you might recommend the query flag as an alternative approach...

File: augur/filter/include_exclude_rules.py [hunk 2]
```diff
@@ -97,10 +97,11 @@ def _parse_filter_query(query):
     ('property', <built-in function ne>, 'value')
 
     """
-    column, value = re.split(r'!?=', query)
-    op = operator.eq
-    if "!=" in query:
-        op = operator.ne
+    split_query = re.split(r'!?=', query)
+    if len(split_query) != 2:
+        raise AugurError(f"Invalid filter query `{query}`. Hint: Queries must follow the pattern of `property=value` or `property!=value`.")
```
Comment (victorlin) [inline augur/filter/include_exclude_rules.py:102]: +1 for @huddlej's suggestion. Example:

```suggestion
        raise AugurError(f"Invalid filter query `{query}`. Hint: Queries must follow the pattern of `property=value` or `property!=value`. `--query` supports more complex queries.")
```

**Review by victorlin — CHANGES_REQUESTED** [review]
